### PR TITLE
crossplane: init at 0.5.7

### PR DESCRIPTION
**Pull request by KAction**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/crossplane/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ lib, buildPythonPackage, fetchFromGitHub }:
+
+buildPythonPackage rec {
+  pname = "crossplane";
+  version = "0.5.7";
+  src = fetchFromGitHub {
```
Comment (fabaff) [inline pkgs/development/python-modules/crossplane/default.nix:6]:
```suggestion
  version = "0.5.7";

  src = fetchFromGitHub {
```

File: pkgs/development/python-modules/crossplane/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    owner = "nginxinc";
+    repo = "crossplane";
+    rev = "v${version}";
+    sha256 = "0lv3frfvnvz5wjxwh3mwy8nbypv4i62v4bvy5fv7vd6kmbxy1q9l";
+  };
+
```
Comment (fabaff) [inline pkgs/development/python-modules/crossplane/default.nix:12]: The upstream repo contains [tests](https://github.com/nginxinc/crossplane/tree/master/tests). 

Also, please add `pythonImportsCheck`.

File: pkgs/development/python-modules/crossplane/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/nginxinc/crossplane";
+    description = "NGINX configuration file parser and builder";
+    licenses = licenses.apsl20;
```
Comment (fabaff) [inline pkgs/development/python-modules/crossplane/default.nix:16]: The license is [Apache Software License](https://github.com/nginxinc/crossplane/blob/master/LICENSE) not the Apple one.

```suggestion
    licenses = licenses.asl20;
```

File: pkgs/development/python-modules/crossplane/default.nix (hunk 2 above)
Comment (KAction) [reply pkgs/development/python-modules/crossplane/default.nix:13]: Done. Any way to get pytest find tests by itself?

File: pkgs/development/python-modules/crossplane/default.nix (hunk 3 above)
Comment (KAction) [reply pkgs/development/python-modules/crossplane/default.nix]: Fixed.

File: pkgs/development/python-modules/crossplane/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/crossplane/default.nix:13]: Normally pytest should find the tests directory by itself.

Can you use the pytestCheckHook?

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -1938,6 +1938,8 @@ in {
 
   croniter = callPackage ../development/python-modules/croniter { };
 
+  crossplane = callPackage ../development/python-modules/crossplane { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:1941]: Do you expect this to be used as a python module? If it is just a script we don't need and want this entry.

File: pkgs/development/python-modules/crossplane/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  version = "0.5.7";
+
+  src = fetchFromGitHub {
+    owner = "nginxinc";
+    repo = "crossplane";
+    rev = "v${version}";
+    sha256 = "0lv3frfvnvz5wjxwh3mwy8nbypv4i62v4bvy5fv7vd6kmbxy1q9l";
+  };
+
+  nativeBuildInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/crossplane/default.nix:14]:
```suggestion
  checkBuildInputs = [ pytest ];
```

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (KAction) [reply pkgs/top-level/python-packages.nix:1941]: It provides couple functions to parse nginx config.
https://pypi.org/project/crossplane/#crossplane-parse

Comment (KAction) [reply pkgs/top-level/python-packages.nix:1941]: I can expect somebody building static analyzer on top of this.

File: pkgs/development/python-modules/crossplane/default.nix (hunk 2 above)
Comment (KAction) [reply pkgs/development/python-modules/crossplane/default.nix:13]: Thanks, that works.

**Comment by KAction** [comment]
@SuperSandro2000 Looks like I addressed everything.

**Pull request merged**